### ISSUE TITLE
Convert Dillinger to a SPA

### ISSUE THREAD

**Issue by joemccann** [joemccann/dillinger#198]
I'm open to ideas on which front end framework to use:  React, Ember, Backbone, Angular.

But a part of me feels it's halfway thru 2014 and we should refactor Dillinger to be a SPA.  When I originally started writing it SPAs were't all that popular/existent so now seems right.

**Comment by Klortho** [comment]
Is this open issue still relevant?

**Comment by joemccann** [comment]
nope!  although if someone made dillinger a react app that would be swell

**Comment by paulocheque** [comment]
I will post some extra feedbacks here that may help about the final decision. 

In #311 I faced Dillinger has 2 main performance issues for big articles:

1) Manual scrolling the Ace editor for the Sync Scroll feature: The implementation is very beautiful and elegant, but using a high value in the Ace Editor `maxLines` attribute results in a bad performance. Using the raw Ace editor scroll event is much faster.

2) Live preview dom update: The `debounce` optimisation is not enough for big articles. I used virtual-dom manually, without any other framework, and the results was much better. But to use a framework, maybe it will be necessary to think in another markdown converter lib.
